### PR TITLE
Add Sparkle hardened runtime support

### PR DESCRIPTION
**Pull request by mbacchi**
This includes the updates:

* Point to Sparkle commit for branch merged_with_sparkle_1.22.0_v2
* Import Python collections library to sign fileop and Autoupdate
  in the correct order

Fixes: https://github.com/brave/brave-browser/issues/6572

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: patches/chrome-installer-mac-signing-signing.py.patch [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
  _PROVISIONPROFILE_DEST = 'embedded.provisionprofile'
-@@ -136,6 +137,7 @@ def get_parts(config):
+@@ -136,7 +138,9 @@ def get_parts(config):
              library_basename.replace('.dylib', ''),
              verify_options=VerifyOptions.DEEP)
  
-+    AddBravePartsForSigning(parts, config)
-     return parts
+-    return parts
++    ordered_parts = collections.OrderedDict(parts)
```
Comment (bridiver) [inline patches/chrome-installer-mac-signing-signing.py.patch]: these additional lines are not needed in this file and should be done internally in `AddBravePartsForSigning`

File: patches/chrome-installer-mac-signing-signing.py.patch [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
-index cd93959385debe65d04c0a4664f638173da54b75..898af6df3915f5770d0e7575eb747c4d14822f13 100644
+index cd93959385debe65d04c0a4664f638173da54b75..70a1d7741908ee9481bd2026ec1d493c086f7e7d 100644
 --- a/chrome/installer/mac/signing/signing.py
 +++ b/chrome/installer/mac/signing/signing.py
-@@ -11,6 +11,7 @@ import os.path
+@@ -8,9 +8,11 @@ bundle that need to be signed, as well as providing utilities to sign them.
+ 
+ import copy
+ import os.path
++import collections
```
Comment (bridiver) [inline patches/chrome-installer-mac-signing-signing.py.patch]: see below

**Comment by mbacchi** [comment]
I've shortened the patch and added logic to the helper function. CC @bridiver

**Comment by mbacchi** [comment]
FYI I rebased after the C80 update on nightly.

**Comment by simonhong** [comment]
I think we should verify this new sparkle version works for updating with our omaha server before merging.

**Comment by bsclifton** [comment]
Since we're having problems with Chromium 80, let's try merging this and doing a build (non-public Nightly). We can always revert if it doesn't work- but we suspect it'll fix the signing problem we've been fighting

**Pull request merged**